src/dxgi/dxgi_adapter.cpp

```diff
@@ -242,13 +242,23 @@ namespace dxvk {
     auto memoryProp = m_adapter->memoryProperties();
     auto deviceId   = m_adapter->devicePropertiesExt().coreDeviceId;
     
-    // Custom Vendor / Device ID
+    // Custom Vendor / Device ID / Description
     if (options->customVendorId >= 0)
       deviceProp.vendorID = options->customVendorId;
     
     if (options->customDeviceId >= 0)
       deviceProp.deviceID = options->customDeviceId;
     
+    if (!options->customDeviceDesc.empty())
+      #ifdef __GNUC__
+      #pragma GCC diagnostic push
+      #pragma GCC diagnostic ignored "-Wstringop-truncation"
+      #endif // __GNUC__
+      std::strncpy(deviceProp.deviceName, options->customDeviceDesc.c_str(), sizeof(deviceProp.deviceName));
```

Comment:
`sizeof(deviceProp.deviceName) - 1` should allow removing the ifdefs

Reply:
Done